config/routes.rb

```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-
-  # Defines the root path route ("/")
-  # root "articles#index"
+  get '/users', to: 'users#index', as: 'users'
+  get '/users/:id', to: 'users#show', as: 'user'
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/:id', to: 'posts#show', as: 'user_post'
```

Comment:
- [ ] [Optioanl] Well! Currently, your implementation looks good. but You can use the `resources` method to achieve a more concise and conventional routing setup. Here's a refactored version of your routes using `resources`:

```ruby
Rails.application.routes.draw do
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
end
```

This will generate the same routes as your original code but in a more compact and readable form. The `only` option is used to specify which actions to include for each resource. The `as` option is used to customize the route helpers.